File: Assignment17/Question4.c
#include <stdio.h> 

void Pattern(int iRow, int iCol) 
{ 
    if (iRow != iCol) 
    { 
        printf("Invalid input\n"); 
        return; 
    } 

    for (int i = 1; i <= iRow; i++) 
    { 
        for (int j = 1; j <= iCol; j++) 
        { 
            if (i == 1 || i == iRow || j == 1 || j == iCol) 
            { 
                printf("*\t"); 
            } 
            else if (
                     (i == 2 && j >= 2 && j <= 4) || 
                     (i == 3 && j >= 2 && j <= 3) || 
                     (i == 4 && j == 2)              
                    )
            { 
                printf("#\t"); 
            } 
            else if (
                     (i == 3 && j == 5) ||           
                     (i == 4 && j >= 4 && j <= 5) || 
                     (i == 5 && j >= 3 && j <= 5)    
                    )
            { 
                printf("$\t"); 
            } 
            else 
            { 
                printf("*\t"); 
            } 
        } 
        printf("\n"); 
    } 
} 

int main() 
{ 
    int iValue1 = 0, iValue2 = 0; 

    printf("Enter number of rows: "); 
    scanf("%d", &iValue1); 

    printf("Enter number of columns: "); 
    scanf("%d", &iValue2); 

    Pattern(iValue1, iValue2); 

    return 0; 
}

File: Assignment51/Question4.java
import java.util.Scanner;
import java.lang.Math;

class MatrixOperations 
{
    public int trace(int[][] matrix, int iRow, int iCol) 
    {
        if (iRow != iCol) 
        {
            System.out.println("Trace is only defined for square matrices.");
            return -1; 
        }
        int sum = 0;
        for (int i = 0; i < iRow; i++) 
        {
            sum = sum + matrix[i][i];
        }
        return sum;
    }

    public double normal(int[][] matrix, int iRow, int iCol) 
    {
        int sumOfAllElements = 0;
        for (int i = 0; i < iRow; i++) 
        {
            for (int j = 0; j < iCol; j++) 
            {
                sumOfAllElements = sumOfAllElements + (matrix[i][j] * matrix[i][j]);
            }
        }
        return Math.sqrt(sumOfAllElements);
    }
}

class Question4 
{
    public static void main(String A[]) 
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter number of rows: ");
        int iRow = sobj.nextInt();

        System.out.print("Enter number of columns: ");
        int iCol = sobj.nextInt();

        int[][] matrix = new int[iRow][iCol];

        System.out.println("Enter elements of the matrix:");
        for (int i = 0; i < iRow; i++) 
        {
            for (int j = 0; j < iCol; j++) 
            {
                matrix[i][j] = sobj.nextInt();
            }
        }
        
        MatrixOperations mobj = new MatrixOperations();

        int traceValue = mobj.trace(matrix, iRow, iCol);
        if (traceValue != -1) 
        {
            System.out.println("\nTrace of the matrix is: " + traceValue);
        }

        double normalValue = mobj.normal(matrix, iRow, iCol);
        System.out.println("Normal of the matrix is: " + normalValue);
    }
}